src/bin/lpython.cpp

```diff
@@ -173,6 +175,51 @@ int emit_asr(const std::string &infile,
         return 2;
     }
     LFortran::ASR::TranslationUnit_t* asr = r.result;
+    for (size_t i=0; i < passes.size(); i++) {
+        switch (passes[i]) {
+            case (ASRPass::do_loops) : {
+                LFortran::pass_replace_do_loops(al, *asr);
+                break;
+            }
+            case (ASRPass::global_stmts) : {
+                LFortran::pass_wrap_global_stmts_into_function(al, *asr, "f");
+                break;
+            }
+            case (ASRPass::implied_do_loops) : {
+                LFortran::pass_replace_implied_do_loops(al, *asr, runtime_library_dir);
+                break;
+            }
+            case (ASRPass::array_op) : {
+                LFortran::pass_replace_array_op(al, *asr, runtime_library_dir);
+                break;
+            }
+            case (ASRPass::inline_function_calls) : {
+                LFortran::pass_inline_function_calls(al, *asr, runtime_library_dir);
+                break;
+            }
+            case (ASRPass::class_constructor) : {
+                LFortran::pass_replace_class_constructor(al, *asr);
+                break;
+            }
+            case (ASRPass::arr_slice) : {
+                LFortran::pass_replace_arr_slice(al, *asr, runtime_library_dir);
+                break;
+            }
+            case (ASRPass::print_arr) : {
+                LFortran::pass_replace_print_arr(al, *asr, runtime_library_dir);
+                break;
+            }
+            case (ASRPass::unused_functions) : {
+              LFortran::pass_unused_functions(al, *asr, true);
+                break;
+            }
+            case (ASRPass::loop_vectorise) : {
+                LFortran::pass_loop_vectorise(al, *asr, runtime_library_dir);
+                break ;
+            }
+            default : throw LFortran::LFortranException("Pass not implemened");
```

Comment:
I think it's time to implement the ASR manager (https://github.com/lcompilers/lpython/issues/656) so that we don't have to maintain such code that is almost (but not quite) identical between LFortran / LPython.

Reply:
I will do that after this.